frontend/src/routes/Governance/discovered/useFetchPolicies.tsx

```diff
@@ -0,0 +1,111 @@
+import { useEffect, useState } from 'react'
+import { grouping } from './grouping'
+import { useRecoilValue, useSharedAtoms } from '../../../shared-recoil'
+import { useSearchResultItemsQuery } from '../../Search/search-sdk/search-sdk'
+import { searchClient } from '../../Search/search-sdk/search-client'
+import { convertStringToQuery } from '../../Search/search-helper'
+import { parseDiscoveryPolicies } from '../common/util'
+export interface ISourceType {
+  type: string //ex: 'Policy' | 'Git' | 'Multiple'
+  parentNs: string
+  parentName: string
+}
+export interface DiscoverdPolicyItem {
+  _hubClusterResource: boolean
+  kind: string
+  apigroup: string
+  apiversion: string
+  cluster: string
+  compliant: string
+  remediationAction: string
+  severity: string
+  _isExternal: boolean
+  annotations: { [annoKey: string]: string }
+  _uid: string
+  created: string
+  label: string
+  kind_plural: string
+  namespace: string
+  name: string
+  disabled: boolean
+  // These are only for Operator policy
+  deploymentAvailable?: boolean | 'yes' | 'no'
+  upgradeAvailable?: boolean | 'yes' | 'no'
+  // Not from search-collector. Attached in grouping function
+  source?: ISourceType
+}
+
+export interface DiscoverdPolicyTableItem {
+  id: string
+  name: string
+  severity: string
+  kind: string
+  responseAction: string
+  policies: DiscoverdPolicyItem[]
+  source?: ISourceType
+}
+
+// If id (`policyName` + `policyKind`) exists, it returns a filtered `DiscoveredPolicyTable` based on `clusterName`.
+export function useFetchPolicies(id?: string) {
+  const [isFetching, setIsFetching] = useState(true)
+  const [data, setData] = useState<DiscoverdPolicyTableItem[]>()
+  const { channelsState, helmReleaseState, subscriptionsState } = useSharedAtoms()
+  const helmReleases = useRecoilValue(helmReleaseState)
+  const subscriptions = useRecoilValue(subscriptionsState)
+  const channels = useRecoilValue(channelsState)
+  const {
+    data: searchData,
+    loading: searchLoading,
+    error: searchErr,
+  } = useSearchResultItemsQuery({
+    client: process.env.NODE_ENV === 'test' ? undefined : searchClient,
+    variables: { input: [convertStringToQuery('kind:ConfigurationPolicy,CertificatePolicy,OperatorPolicy', 100000)] },
+  })
+  const searchDataItems = searchData?.searchResult?.[0]?.items
+
+  useEffect(() => {
+    setIsFetching(true)
+    if (searchDataItems && !searchErr && !searchLoading) {
+      const dataObj = '(' + grouping + ')();'
+      // for firefox
+      const blob = new Blob([dataObj.replace('"use strict";', '')], { type: 'application/javascript' })
+      const blobURL = (window.URL ? URL : webkitURL).createObjectURL(blob)
+      // Worker for discovered policies table
+      const worker = new Worker(blobURL)
+
+      worker.onmessage = (e: MessageEvent<any>) => {
+        id
+          ? setData((parseDiscoveryPolicies(e.data) as DiscoverdPolicyTableItem[]).filter((item) => item.id == id))
```

Comment:
I understand we need the ID for the table, but I'd prefer if `.name` and `.kind` were compared directly to make the code clearer.

Reply:
we need to compare `.name ` && `.kind` && `.apiGroup`... What if we make id like this "{name}-{kind}-{apiGroup}"

Also do we have to include `apiVersion` in `id`?

Reply:
API group or API version is okay. We just need one and not the other. The reason being that search only returns one version for each resource (the one that is set to `storage: true` in the CRD) so there's no need to differentiate the version.